### PR TITLE
[Merged by Bors] - Add NodeID to events

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation

This adds the NodeID of the Smesher that created that event to various events.

## Description

Merge after https://github.com/spacemeshos/api/pull/314

Adds the identity that emits the event to various events.

## Test Plan

N/A

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: events/events.go [hunk 1]
```diff
@@ -24,53 +24,55 @@ func EmitBeacon(epoch types.EpochID, beacon types.Beacon) {
 		&pb.Event_Beacon{
 			Beacon: &pb.EventBeacon{
 				Epoch:  epoch.Uint32(),
-				Beacon: beacon[:],
+				Beacon: beacon.Bytes(),
```
Comment (dshulyak) [inline events/events.go:27]: does it copy only 5 bytes? because there are some _smart_ methods that will cast value to Hash32 first and then return bytes

**Review by dshulyak — APPROVED** [review]

File: events/events.go (hunk 1 above)
Comment (fasmat) [reply events/events.go:27]: the method is exactly the same:

```go
func (b Beacon) Bytes() []byte {
	return b[:]
}
```

I mainly did it for consistency (some use `[:]` some use `.Bytes()` in the same file). I checked others to be sure:

- `NodeID` is same as Beacon (just `[:]`)
- `ATXID` first casts to `Hash32` but `ATXID` is a `Hash32` so this should be free (?)

Comment (dshulyak) [reply events/events.go:27]: yes, i think the faulty are Ballot and Proposal, which are 20 bytes but then casted to 32 in Bytes() method

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**